rotkehlchen/accounting/history_base_entries.py

```diff
@@ -196,33 +211,30 @@ def _process_swap(
                 fee_taxable = True
                 fee_taxable_amount_ratio = trade_taxable_amount / out_event.balance.amount
 
-            self.pot.add_out_event(
-                event_type=AccountingEventType.FEE,
-                notes=fee_event.notes,  # type: ignore [arg-type]  # notes exist here
-                location=fee_event.location,
-                timestamp=timestamp,
-                asset=fee_event.asset,
-                amount=fee_event.balance.amount,
-                taxable=fee_taxable,
-                given_price=fee_price,
-                # By setting the taxable amount ratio we determine how much of the fee
-                # spending should be a taxable spend and how much free.
-                taxable_amount_ratio=fee_taxable_amount_ratio,
-                count_cost_basis_pnl=True,
-                count_entire_amount_spend=True,
-                extra_data=extra_data,
-            )
-            consumed_events = 3
-
-        self.pot.add_in_event(
-            event_type=AccountingEventType.TRANSACTION_EVENT,
-            notes=in_event.notes if in_event.notes else '',
-            location=in_event.location,
-            timestamp=timestamp,
-            asset=in_event.asset,
-            amount=in_event.balance.amount,
-            taxable=False,  # acquisitions in swaps are never taxable
-            given_price=prices[1],
-            extra_data=extra_data,
-        )
-        return consumed_events
+            events_to_add_queue.extend([
+                ('add_in_event', add_in_event_kwargs),
+                ('add_out_event', {
```

Comment:
no need for dynamic names. It's constant

```suggestion
                (self.pot.add_in_event, add_in_event_kwargs),
                (self.pot.add_out_event, {
```